src/core/kernel/inspect.rs

```diff
@@ -231,37 +265,38 @@ pub(super) fn function_nargs(symbol: &Symbol) -> Result<u32> {
     };
 
     let proto = get_function_prototype(symbol)?;
+
     Ok((proto.parameters.len() - fix) as u32)
 }
 
 fn get_function_prototype(symbol: &Symbol) -> Result<btf_rs::FuncProto> {
-    let btf = &get_inspector!()?.btf;
     let target = symbol.typedef_name();
+    let (btf, func) = &get_inspector!()?.btf_find(|b| b.resolve_type_by_name(&target))?;
```

Comment:
Note this is still WIP so this implementation is just a demonstration of the btf-rs PR.

> The logic is to return a reference to the Btf obj where a symbol was found, so subsequent requests can be made using the same module as an optimization. Will that work when modules have a dependency on another module? I'm thinking types, struct, etc would be described in both but is that true for everything?

It's actually more than just a matter of optimization.

If a specific type is resolved by name, we can iterate though all modules to try to find it. When we find it, and then try to resolve a chained or member type by ID, then we _must_ use the same `Btf` file because IDs can be duplicated in multiple modules.

Looking back at this, I'm thinking we should not open `btf_find` to any query (because using it with `resolve_by_id` can be potentially wrong). So for starters we should refactor `get_function_prototype` and `is_param_type` to use the same `Btf` object (the one found by looking into all modules by name first)